src/Network/ListenerHandler.cs

```diff
@@ -110,16 +110,15 @@ public async void HandleConnection(object sender, IConnection connection)
                         Diagnostic.Debug($"Distributed PierceFirewall with token {pierceFirewall.Token} received from {distributedUsername} ({connection.IPEndPoint.Address}:{SoulseekClient.Listener.Port}) (id: {connection.Id})");
                         SoulseekClient.Waiter.Complete(new WaitKey(Constants.WaitKey.SolicitedDistributedConnection, distributedUsername, pierceFirewall.Token), connection);
                     }
-                    else if (SoulseekClient.Options.SearchResponseCache != default)
+                    else if (SoulseekClient.Options.SearchResponseCache != null && SoulseekClient.Options.SearchResponseCache.TryGet(pierceFirewall.Token, out var cachedSearchResponse))
```

Comment:
The copy of msbuild on circle CI was really unhappy about the null coalescing operator here, but I couldn't reproduce it locally.  Splitting the null check fixes it.